src/OfflineWorkflowsSample/OfflineWorkflowsSample/ViewModels/LoginViewModel.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Esri.ArcGISRuntime.Http;
+using Esri.ArcGISRuntime.Portal;
+using Esri.ArcGISRuntime.Security;
+using OfflineWorkflowsSample;
+using OfflineWorkflowsSample.Models;
+using Prism.Commands;
+using Prism.Windows.Mvvm;
+
+namespace OfflineWorkflowSample.ViewModels
+{
+    public class LoginViewModel : ViewModelBase
+    {
+        // Commands enable binding controls to behavior. https://visualstudiomagazine.com/articles/2012/04/10/command-pattern-in-net.aspx
+        private readonly DelegateCommand _logInToEnterpriseCommand;
+        private readonly DelegateCommand _logInToPortalCommand;
+        private readonly DelegateCommand _showEnterpriseFormCommand;
+
+        private bool _portalFormOpen;
+
+        private string _portalUrl = ArcGISOnlinePortalUrl;
+        
+        // WindowService allows the ViewModel to communicate with the view without
+        //     exposing details of the view to the ViewModel. 
+        public IWindowService WindowService = null;
```

Comment:
As far as I can tell, WindowService always stays null. So whenever this window tries to display an error, app dies with a NullReferenceException.

Reply:
Doesn't show on github but I pushed a fix to set the window service in the page's constructor.